### PR TITLE
Trajopt clang-tidy clean-up v3

### PR DESCRIPTION
**Pull request by rjoomen**

### CODE REVIEW

**Comment by rjoomen** [comment]
Who'd have thought [retrofitting const](https://isocpp.org/wiki/faq/const-correctness#retrofitting-const) is such a lot of work.

**Comment by Levi-Armstrong** [comment]
> Who'd have thought [retrofitting const](https://isocpp.org/wiki/faq/const-correctness#retrofitting-const) is such a lot of work.

Yea, this is one thing I had intentionally avoided doing because it can have performance impact that was not checked by the older version of clang-tidy, but now it will produce an error if you return a variable which was previous defined const.

File: trajopt_ifopt/src/constraints/cartesian_position_constraint.cpp [hunk 1]
```diff
@@ -193,27 +193,27 @@ CartPosConstraint::CartPosConstraint(CartPosInfo info,
   }
 }
 
-CartPosConstraint::CartPosConstraint(CartPosInfo info,
+CartPosConstraint::CartPosConstraint(const CartPosInfo& info,
                                      std::shared_ptr<const JointPosition> position_var,
                                      const std::string& name)
-  : CartPosConstraint(std::move(info), std::move(position_var), Eigen::VectorXd::Ones(info.indices.rows()), name)
+  : CartPosConstraint(info, std::move(position_var), Eigen::VectorXd::Ones(info.indices.rows()), name)
```
Comment (Levi-Armstrong) [inline trajopt_ifopt/src/constraints/cartesian_position_constraint.cpp:199]: Good catch, did clang-tidy find this issue?

File: trajopt_ifopt/src/costs/absolute_cost.cpp [hunk 2]
```diff
@@ -32,8 +32,8 @@ TRAJOPT_IGNORE_WARNINGS_POP
 
 namespace trajopt_ifopt
 {
-AbsoluteCost::AbsoluteCost(ifopt::ConstraintSet::Ptr constraint)
-  : AbsoluteCost(std::move(constraint), Eigen::VectorXd::Ones(constraint->GetRows()))
+AbsoluteCost::AbsoluteCost(const ifopt::ConstraintSet::Ptr& constraint)
+  : AbsoluteCost(constraint, Eigen::VectorXd::Ones(constraint->GetRows()))
 {
 }
```
Comment (Levi-Armstrong) [inline trajopt_ifopt/src/costs/absolute_cost.cpp:38]: good catch!

File: trajopt_ifopt/src/costs/squared_cost.cpp [hunk 3]
```diff
@@ -32,8 +32,8 @@ TRAJOPT_IGNORE_WARNINGS_POP
 
 namespace trajopt_ifopt
 {
-SquaredCost::SquaredCost(ifopt::ConstraintSet::Ptr constraint)
-  : SquaredCost(std::move(constraint), Eigen::VectorXd::Ones(constraint->GetRows()))
+SquaredCost::SquaredCost(const ifopt::ConstraintSet::Ptr& constraint)
+  : SquaredCost(constraint, Eigen::VectorXd::Ones(constraint->GetRows()))
 {
 }
```
Comment (Levi-Armstrong) [inline trajopt_ifopt/src/costs/squared_cost.cpp:38]: good catch!

**Review by Levi-Armstrong — APPROVED** [review]

File: trajopt_ifopt/src/constraints/cartesian_position_constraint.cpp (hunk 1 above)
Comment (rjoomen) [reply trajopt_ifopt/src/constraints/cartesian_position_constraint.cpp:199]: Yes, clang-tidy is great for that!

**Pull request merged**